docs/cli.md

```diff
@@ -64,54 +68,60 @@ The general structure for running commands with the Gravitino CLI is `gcli entit
  -x,--index              display index information
  -z,--provider <arg>     provider one of hadoop, hive, mysql, postgres,
                          iceberg, kafka
- ```
+```
 
-## Commands
+## Commands Types
 
 The following commands are used for entity management:
 
-- list: List available entities
-- details: Show detailed information about an entity
-- create: Create a new entity
-- delete: Delete an existing entity
-- update: Update an existing entity
-- set: Set a property on an entity
-- remove: Remove a property from an entity
-- properties: Display an entities properties
+- `create`: Create a new entity
+- `delete`: Delete an existing entity
+- `details`: Show detailed information about an entity
+- `list`: List available entities
+- `properties`: Display an entity's properties
+- `remove`: Remove a property from an entity
+- `set`: Set a property on an entity
+- `update`: Update an existing entity
```

Comment:
Please leave the order as it, it ordered according to use of command / logical grouping. The list and details commands belong together, so do create/delete/update and set/remove. Alphabetical order is not helpful here.

Reply:
As I have responded in another review comment. When we don't have a consensus on the
ordering of things, the best way to do it is to sort them in alphabetic order.
Why? Every people may have a different opinion on the definition of "logical grouping".
For example, why the first entry should be `list`? What am I doing this if I haven't
created something? Why `delete` comes before `update`?
The main problem of this "logical grouping" is about predictability. There will never
be a consensus on the correct/golden ordering.
With that, I'm strongly against it.

Reply:
While there may be other logical ways to order them, that doesn't mean we should do it alphabetically. `list` is first as it is most likely the first command a user will want to use, followed by `details` the next most likely command a user would use and `list` and `details` are commonly used together. `delete` pairs with `create` so that is why it comes before `update`. If the commands were slightly different, I'd would have gone with a more CRUD style ordering.

As a native English speaker, I would not expect this to be in alphabetical order. Alphabetical order is used when you have a long list. If there were, say 20 options, I would sort them alphabetically or break them up into groups. Not in the `--help` command is sorted alphabetically, but I had considered following this advice from https://clig.dev/#help 

"Display the most common flags and commands at the start of the help text. It’s fine to have lots of flags, but if you’ve got some really common ones, display them first. For example, the Git command displays the commands for getting started and the most commonly used subcommands first:"

There is also actual usability research on this:
https://www.nngroup.com/articles/alphabetical-sorting-must-mostly-die/

Basically, when users don't know the name of the thing they want, it makes A–Z listings useless and A–Z listings are directly harmful because they hide logical ordering.

Reply:
@justinmclean That is persuasive. I was most of the time treating this docs as reference
rather than a tutorial. In a tutorial or guideline, we leave the least frequently used
(aka, "advanced") commands at the bottom of the list.
In this case, maybe, as you said, we can keep the short list as is.